controllers/volumereplicationgroup_controller.go

```diff
@@ -197,19 +201,34 @@ func (v *VolumeReplicationGroupReconciler) Reconcile(ctx context.Context, req ct
 
 	pvcList, err := v.HandlePersistentVolumeClaims(ctx, volRepGroup)
 	if err != nil {
-		log.Error("Handling of Persistent Volume Claims of application failed: ", volRepGroup.Spec.ApplicationName)
+		log.Error(err, "Handling of Persistent Volume Claims of application failed")
 
 		return ctrl.Result{}, err
 	}
 
-	err = v.HandlePersistentVolumes(ctx, pvcList)
-	if err != nil {
-		log.Error("Handling of Persistent Volumes for PVCs of application failed: ", volRepGroup.Spec.ApplicationName)
+	requeue := false
 
-		return ctrl.Result{}, err
+	for _, pvc := range pvcList.Items {
+		volRep := &volrep.VolumeReplication{}
+
+		err = v.Get(ctx, types.NamespacedName{Name: pvc.Name, Namespace: pvc.Namespace}, volRep)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				if err = v.createVolumeReplicationCRForPVC(ctx, pvc); err == nil {
+					continue
+				}
+			}
+
+			// requeue on failure to ensure any PVC not having a corresponding VR CR
+			requeue = true
```

Comment:
Success of get and create was tested, error condition remains untested to ensure re-queue.